plugins/ledger/auth2fa.py

```diff
@@ -118,9 +120,17 @@ def return_pin():
         def pin_changed(s):
             if len(s) < len(self.idxs):
                 i = self.idxs[len(s)]
-                addr = self.txdata['address']
-                addr = addr[:i] + '<u><b>' + addr[i:i+1] + '</u></b>' + addr[i+1:]
-                self.addrtext.setHtml(str(addr))
+                address = self.txdata['address']
+
+                # Always generate the mainnet address as the code is generated from mainnet address
+                addressstr = address.to_string(Address.FMT_LEGACY, net=networks.MainNet)
+                addressstr = addressstr[:i] + '<u><b>' + addressstr[i:i+1] + '</u></b>' + addressstr[i+1:]
+
+                # We also show the UI address if it is different
+                if networks.net.TESTNET or not Address.FMT_UI == Address.FMT_LEGACY:
```

Comment:
Not sure what's going on here but I think I understand.  Ok never mind.  I'll just merge.

Reply:
Yea, I've been thinking of changing that yesterday as it is more difficult to understand than it should be. More clear would be to compare the output of `address.to_string(Address.FMT_LEGACY, net=networks.MainNet)` to the output of `address.to_ui_string()` to determine whether to display both addresses.